stream/writer.go

```diff
@@ -0,0 +1,36 @@
+package stream
+
+type NodeWriter interface {
```

Comment:
What do you mean? Merge the `stream` and `codec` packages? Perhaps, but this will make a huge package, and the two can live independently. Also, there may be issues with circular package dependencies (mostly in the test files).